### PR TITLE
RandomScan to LuraToon

### PR DESCRIPTION
**Pull request by domflamingo2**
Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [ ] Have tested the modifications by compiling and running the extension through Android Studio
- [X] Have removed `web_hi_res_512.png` when adding a new extension
- [X] Change domain 

Random Scan to LuraToon (pt-BR): https://randomscanlators.net -> https://luratoon.com/
Closes (#500)

### CODE REVIEW

File: multisrc/overrides/madara/randomscan/src/luratoon.kt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package eu.kanade.tachiyomi.extension.pt.randomscan
+package eu.kanade.tachiyomi.extension.pt.luratoon
```
Comment (beerpiss) [inline multisrc/overrides/madara/randomscan/src/luratoon.kt:1]: Preserve the package name so users get updates through the app.

Comment (beer-psi) [inline multisrc/overrides/madara/randomscan/src/luratoon.kt]: Preserve the old package name so users get updates through the app.

File: multisrc/overrides/madara/randomscan/src/luratoon.kt [hunk 2]
```diff
@@ -7,9 +7,9 @@ import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class RandomScan : Madara(
-    "Random Scan",
-    "https://randomscanlators.net",
+class LuraToon : Madara(
+    "LuraToon",
```
Comment (beer-psi) [inline multisrc/overrides/madara/randomscan/src/luratoon.kt]: When doing a name change, you need to preserve the old source's `id` so library entries can be migrated automatically.

Look for the source's `id` [here](https://raw.githubusercontent.com/keiyoushi/extensions/repo/index.json).

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/madara/MadaraGenerator.kt [hunk 3]
```diff
@@ -415,7 +415,7 @@ class MadaraGenerator : ThemeSourceGenerator {
         SingleLang("RagnarokScan", "https://ragnarokscan.com", "es", overrideVersionCode = 1),
         SingleLang("Raijin Scans", "https://raijinscans.fr", "fr"),
         SingleLang("Rainbow Fairy Scan", "https://rainbowfairyscan.com", "pt-BR"),
-        SingleLang("Random Scan", "https://randomscanlators.net", "pt-BR", overrideVersionCode = 6),
+        SingleLang("Lura Toon", "https://luratoon.com/", "pt-BR", overrideVersionCode = 7),
```
Comment (beer-psi) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/madara/MadaraGenerator.kt]:
```suggestion
        SingleLang("Lura Toon", "https://luratoon.com/", "pt-BR", overrideVersionCode = 7, pkgName = "randomscan", className = "LuraToon"),
```

File: multisrc/overrides/madara/randomscan/src/luratoon.kt [hunk 4]
```diff
@@ -7,9 +7,9 @@ import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class RandomScan : Madara(
-    "Random Scan",
-    "https://randomscanlators.net",
+class LuraToon : Madara(
```
Comment (beer-psi) [inline multisrc/overrides/madara/randomscan/src/luratoon.kt]: Match the class' name with the filename, preserving casing. The filename should have been `LunaToon.kt`.

**Review by beer-psi — CHANGES_REQUESTED** [review]

**Review by beer-psi — CHANGES_REQUESTED** [review]
Also, use a specific format when [linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) so that the issue gets automatically closed when the PR merges.

File: multisrc/overrides/madara/randomscan/src/luratoon.kt (hunk 1 above)
Comment (domflamingo2) [reply multisrc/overrides/madara/randomscan/src/luratoon.kt]: done

Comment (domflamingo2) [reply multisrc/overrides/madara/randomscan/src/luratoon.kt]: done

File: multisrc/overrides/madara/randomscan/src/luratoon.kt (hunk 4 above)
Comment (domflamingo2) [reply multisrc/overrides/madara/randomscan/src/luratoon.kt]: why "LunaToon.kt" if the site has changed to LuraToon?

Comment (beer-psi) [reply multisrc/overrides/madara/randomscan/src/luratoon.kt]: > why "LunaToon.kt" if the site has changed to LuraToon?

That was a typo. The point is that for single-class files, the filename should be `{class name}.kt`. In this case, `LuraToon.kt`.

Comment (domflamingo2) [reply multisrc/overrides/madara/randomscan/src/luratoon.kt]: okay then, I'll change the file name.

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/madara/MadaraGenerator.kt [hunk 5]
```diff
@@ -415,7 +415,7 @@ class MadaraGenerator : ThemeSourceGenerator {
         SingleLang("RagnarokScan", "https://ragnarokscan.com", "es", overrideVersionCode = 1),
         SingleLang("Raijin Scans", "https://raijinscans.fr", "fr"),
         SingleLang("Rainbow Fairy Scan", "https://rainbowfairyscan.com", "pt-BR"),
-        SingleLang("Random Scan", "https://randomscanlators.net", "pt-BR", overrideVersionCode = 6),
+        SingleLang("Lura Toon", "https://luratoon.com/", "pt-BR", overrideVersionCode = 7, pkgName = "randomscan", className = "LuraToon"),
```
Comment (AwkwardPeak7) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/madara/MadaraGenerator.kt]:
```suggestion
        SingleLang("Lura Toon", "https://luratoon.com", "pt-BR", overrideVersionCode = 7, pkgName = "randomscan", className = "LuraToon"),
```

File: multisrc/overrides/madara/randomscan/src/luratoon.kt [hunk 6]
```diff
@@ -7,9 +7,9 @@ import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class RandomScan : Madara(
+class luratoon : Madara(
     "Random Scan",
-    "https://randomscanlators.net",
+    "https://luratoon.com/",
```
Comment (AwkwardPeak7) [inline multisrc/overrides/madara/randomscan/src/luratoon.kt]:
```suggestion
    "https://luratoon.com",
```
remove trailing slash

File: multisrc/overrides/madara/randomscan/src/LunaToon.kt [hunk 7]
```diff
@@ -7,9 +7,9 @@ import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class RandomScan : Madara(
+class LunaToon : Madara(
```
Comment (beer-psi) [inline multisrc/overrides/madara/randomscan/src/LunaToon.kt]: You need to match the class name with the `className` you have declared in the generator.

If in `MadaraGenerator.kt` you wrote this:
```kotlin
SingleLang(/* snipped */ className = "LuraToon"),
```

Then the class in overrides must be `LuraToon` and the file name must be `LuraToon.kt`. Please actually understand what you're trying to do and test your changes properly, or any of your future pull requests will be automatically rejected.

**Review by beer-psi — CHANGES_REQUESTED** [review]

File: multisrc/overrides/madara/randomscan/src/LunaToon.kt
Comment (AwkwardPeak7) [inline multisrc/overrides/madara/randomscan/src/LunaToon.kt:1]: their name is "Lura Toon" why are you naming the classs and file "Luna Toon"?

File: multisrc/overrides/madara/randomscan/src/LunaToon.kt [hunk 8]
```diff
@@ -7,9 +7,9 @@ import java.text.SimpleDateFormat
...
     "Random Scan",
```
Comment (AwkwardPeak7) [inline multisrc/overrides/madara/randomscan/src/LunaToon.kt]: change name here like you did in the generator, also need to override the id as beer asked earlier.

**Review by AwkwardPeak7 — DISMISSED** [review]

File: multisrc/overrides/madara/randomscan/src/LunaToon.kt (hunk 8 above)
Comment (domflamingo2) [reply multisrc/overrides/madara/randomscan/src/LunaToon.kt]: Done, I changed the name of the file to LuraToon.kt and the class to LuraToon, according to MadaraGeneretor in "className = "LuraToon""

Comment (domflamingo2) [reply multisrc/overrides/madara/randomscan/src/LunaToon.kt]: but anything?, before making the commit?

Comment (beer-psi) [reply multisrc/overrides/madara/randomscan/src/LunaToon.kt]: > but anything?, before making the commit?

as AwkwardPeak said, you need to:
- Change the source name: `"Random Scan"` -> `"Lura Toon"`
- Override the ID: find the old source ID [here](https://raw.githubusercontent.com/keiyoushi/extensions/repo/index.json) and see how to override the ID [here](https://github.com/search?q=repo%3Akeiyoushi%2Fextensions-source+%22override+val+id%22&type=code).

Comment (domflamingo2) [reply multisrc/overrides/madara/randomscan/src/LunaToon.kt]: // Name changed from Random Scan to Lura Toon
`override val id: Long = 8295593204533043008` ?


```
class LuraToon : Madara(
    "Lura Toon",
    "https://luratoon.com",
    "pt-BR",
    SimpleDateFormat("MMMMM dd, yyyy", Locale("pt", "BR")),
```

**Review by beer-psi — APPROVED** [review]

**Pull request merged**